parsl/dataflow/dflow.py

```diff
@@ -598,20 +558,15 @@ def cleanup(self):
         # Send final stats
         self.usage_tracker.send_message()
         self.usage_tracker.close()
-        # We do not need to cleanup if the executors are managed outside
-        # the DFK
-        if not self._executors_managed:
-            return
 
         logger.info("Terminating flow_control and strategy threads")
         self.flowcontrol.close()
 
         for executor in self.executors.values():
-            if executor.scaling_enabled:
-                job_ids = executor.execution_provider.resources.keys()
+            if executor.scaling_enabled and executor.managed:
+                job_ids = executor.provider.resources.keys()
                 executor.scale_in(len(job_ids))
 
-            # We are not doing shutdown here because even with block=False this blocks.
             executor.shutdown()
```

Comment:
We shouldn't call executor shutdown on non managed executors. Just fix indent here and if block